benchmarking/runs/csv_weight_run_revised/weight_run_revised.py

```diff
@@ -0,0 +1,159 @@
+            ["ResponseId", "Q8", "Q4", "Q5", "zPos", "TeamSizeViolation", "TeamId"]
+        ]
+        for team in team_set.teams:
+            for student in team.students:
+                attributes = student.attributes
+
+                unique_id = student_provider.get_student(student.id)
+                time_slot = attributes[ScenarioAttribute.TIMESLOT_AVAILABILITY.value][0]
+                tutor_preference = attributes[Attributes.TUTOR_PREFERENCE.value][0]
+                group_size = attributes[Attributes.GROUP_SIZE.value][0]
+
+                data_fields_input_1 = Attributes.revert_timeslot(time_slot)
+                data_fields_input_2 = Attributes.revert_tutor_preference(
+                    tutor_preference
+                )
+                data_fields_input_3 = Attributes.revert_group_size(group_size)
+
+                # Process score and team size violation
+                positive_z = "1" if attributes[Attributes.SCORE.value][0] == 1 else "0"
+                num_teams = len(team.students)
+
+                # Can be customize to fit the team size preference
+                team_size_violation = (
+                    "Yes"
+                    if tutor_preference == 3
+                    and (
+                        (group_size == 1 and num_teams != 2)
+                        or (group_size == 2 and (num_teams > 4 or num_teams < 3))
+                        or (group_size == 3 and num_teams < 4)
+                    )
+                    else ""
+                )
+
+                data_fields.append(
+                    [
+                        unique_id,
+                        data_fields_input_1,
+                        data_fields_input_2,
+                        data_fields_input_3,
+                        positive_z,
+                        team_size_violation,
+                        team.id,
+                    ]
+                )
+
+        # Directory where the new CSV file will be written
+        output_dir = path.join(path.dirname(__file__), "new_csv_output")
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+
+        output_file = os.path.join(output_dir, "result.csv")
+
+        with open(output_file, "w+", newline="") as f:
+            writer = csv.writer(f)
+            for row in data_fields:
+                writer.writerow(row)
+
+
+class DefaultScenario(Scenario):
+    @property
+    def name(self) -> str:
+        return "Diversify on score and concentrate on timeslot"
+
+    @property
+    def goals(self) -> List[Goal]:
+        """
+        This function is meant to create a list of scenerio goals to be followed for team formation.
+
+        Returns:
+            List[Goal]: A list of goals for team formation.
+        """
+        return [
+            # Concentrate on student timeslot availability
+            DiversityGoal(
+                DiversifyType.CONCENTRATE,
+                ScenarioAttribute.TIMESLOT_AVAILABILITY.value,
```

Comment:
Applies to every other file too, but it's dangerous to be using ScenarioAttribute at all in any of this